### PR TITLE
Preferences: add custom icon setting

### LINKED ISSUE #16: Custom Icons?
Will you be adding support to upload custom icons?

### PR DESCRIPTION
**Pull request by halfmexican**
Closes #16 

image

I not entirely sure if I'm happy with the current UI, but this implementation works. I'd appreciate any suggestions and feedback.

### CODE REVIEW

**Comment by Aryan20** [comment]
Thanks! I will check the implementation and will think of something in terms of UI soon.

**Comment by halfmexican** [comment]
image

**Comment by halfmexican** [comment]
I think I want to add a icon preview next to the button

**Comment by Aryan20** [comment]
Hey! Any updates?

**Review by Aryan20 — CHANGES_REQUESTED** [review]

**Comment by Aryan20** [comment]
Finally got GitHub to make the comments public lol

**Comment by halfmexican** [comment]
life has gotten in the way.

**Comment by Aryan20** [comment]
> life has gotten in the way.

Ah okay, no issues, take your time and happy holidays :)

File: PrefsLib/adw.js [hunk 1]
```diff
@@ -150,10 +152,67 @@ export const LogoMenuIconsPage = GObject.registerClass(class LogoMenuIconsWidget
...
+        });
+
+        customIconSelectionRow.add_suffix(customIconButton);
+        customIconRow.add_row(customIconSelectionRow);
+
         // iconGroup
         symbolicIconGroup.add(symbolicIconsRow);
         colouredIconGroup.add(colouredIconsRow);
         iconSettingsGroup.add(menuButtonIconSizeRow);
+        iconSettingsGroup.add(customIconRow);
```
[the remark this answers is not in the document]
Comment (halfmexican) [reply PrefsLib/adw.js:215]: This sounds completely reasonable, will implement

**Review by Aryan20 — APPROVED** [review]
LGTM!

**Comment by Aryan20** [comment]
Thanks for working on this :D

**Pull request merged**